src/BoilerPlate/Asteroid.cpp

```diff
@@ -1,23 +1,38 @@
 #include "Asteroid.hpp"
-#include "Vector2.hpp"
+
 #include "MathUtilities.hpp"
+#include "Vector2.hpp"
+
 #include <GL/glew.h>
 #include <SDL2/SDL_opengl.h>
 
 #include <vector>
 #include <cmath>
-#include <ctime>
-#include <cstdio>
+
+const int MAX_SPEED = 500;
 
 std::vector <Vector2> asteroidPoints;
 
+inline float getSizeFactor(AsteroidSize size) {
+	return 2.0f*static_cast<int>(size) + 1;
+}
 
-Asteroid::Asteroid (AsteroidSize _size, int width, int height)
-	: SpaceObject(width,height), m_asteroidSize(_size)
+Asteroid::Asteroid (AsteroidSize size, int width, int height)
+	: SpaceObject(width, height), m_asteroidSize(size)
 {
-	position = new Vector2(MathUtilities::RandInt(width)-width/2, MathUtilities::RandInt(height) - height/2);
-	m_rotAng = MathUtilities::Interpolate(0.0f, MathUtilities::PI * 2.0f, MathUtilities::RandFloat());
+	float sizeFactor = getSizeFactor(size);
```

Comment:
Remember to be consistent with the naming. For instance, you're using camel case for getSizeFactor, but PascalCase for Interpolate?